hail_search/test_search.py

```diff
@@ -245,6 +245,29 @@ async def test_frequency_filter(self):
             omit_sample_type='SV_WES',
         )
 
+    async def test_annotations_filter(self):
+        await self._assert_expected_search([VARIANT2], pathogenicity={'hgmd': ['hgmd_other']}, omit_sample_type='SV_WES')
+
+        pathogenicity = {'clinvar': ['likely_pathogenic', 'vus_or_conflicting', 'benign']}
+        await self._assert_expected_search([VARIANT2], pathogenicity=pathogenicity, omit_sample_type='SV_WES')
+
+        await self._assert_expected_search(
+            [VARIANT2, VARIANT4], pathogenicity=pathogenicity, annotations={'SCREEN': ['CTCF-only', 'DNase-only']},
+            omit_sample_type='SV_WES',
+        )
+
+        annotations = {'missense': ['missense_variant'], 'in_frame': ['inframe_insertion', 'inframe_deletion'], 'frameshift': None}
+        await self._assert_expected_search(
+            [VARIANT2, VARIANT4], pathogenicity=pathogenicity, annotations=annotations, omit_sample_type='SV_WES',
+        )
+
+        await self._assert_expected_search([VARIANT2, VARIANT4], annotations=annotations, omit_sample_type='SV_WES')
```

Comment:
Are there special cases covered with and without the `pathogenicity` in these two tests?

Reply:
There are 6 different assertions in this one new test, so I don't know which 2 tests you are referring to, but this test has multiple cases with pathogenicity only, annotations only, and both annotations and pathogenicity. What is the actual case you are concerned about?

Reply:
The `pathogenicity` could override the `annotations`. But the search results are the same. So the `pathogenicity` and `annotations` are overlapped. The concern is the test efficiency.